lib/src/behaviors/behavior.dart

```diff
@@ -21,4 +21,9 @@ abstract class Behavior<Parent extends Entity> extends Component
     assert(component is! Behavior, 'Behaviors cannot have behaviors.');
     return super.add(component);
   }
+
+  @override
+  bool containsPoint(Vector2 point) {
```

Comment:
Why is this an API that's available on all behaviors?

Reply:
The reason I put it on the base one is simply because behaviors have (or will have) an inherited size and position from their parent entity. Any Flame functionality that wants to confirm if a given component is containing a point expects this method to be correctly implement but the default `Component` from which we extend has a `containsPoint` that returns false. 

So we just proxy this method, same as we will do for `position`, `size`, `angle` and others that are coming from the parent.

Reply:
From the docs:
![image](https://user-images.githubusercontent.com/15887627/168784430-e8c59207-f1bb-4753-aa07-1cdd0434293d.png)

Reply:
Should this be a specialized subclass then? not all behaviors should semantically able to have a visual representation, right? 
Ex: a "TimerBehavior" or a "KeyboardHandlerBehavior".

Reply:
To be clear I also disagree with myself. In this package, Entitites are always a position component.